src/NServiceBus.TransactionalSession/TransactionalSessionDelayControlMessageBehavior.cs

```diff
@@ -19,16 +19,15 @@ public TransactionalSessionDelayControlMessageBehavior(IMessageDispatcher dispat
 
         public override async Task Invoke(IIncomingPhysicalMessageContext context, Func<Task> next)
         {
-            var isCommitControlMessage = context.Message.Headers.ContainsKey(OutboxTransactionalSession.RemainingCommitDurationHeaderName);
-
-            if (isCommitControlMessage == false)
+            if (!context.Message.Headers.TryGetValue(OutboxTransactionalSession.RemainingCommitDurationHeaderName, out var remainingCommitDurationHeader)
```

Comment:
I'm wondering if we should have a dedicated header to identify the control message instead of relying on these two to be there.

Reply:
do you see a scenario where we send this control message without these headers? And a scenario where the control message can be used without both of these values being present?